File: oxian_core/src/models/interface.rs
#[derive(Debug, Clone, PartialEq, Eq)]
pub enum InterfaceStatus {
    Up,
    Down,
    Testing,
    Unknown(u32),
}

impl From<u32> for InterfaceStatus {
    fn from(value: u32) -> Self {
        match value {
            1 => Self::Up,
            2 => Self::Down,
            3 => Self::Testing,
            v => Self::Unknown(v),
        }
    }
}

#[derive(Debug, Clone, PartialEq)]
pub struct Interface {
    pub index: u32,
    pub description: Option<String>,
    pub mac_address: Option<String>,
    pub admin_status: Option<InterfaceStatus>,
    pub oper_status: Option<InterfaceStatus>,
}

File: oxian_py/models/interface.py
from __future__ import annotations

from enum import Enum
from pydantic import BaseModel, ConfigDict


class InterfaceStatus(str, Enum):
    """Operational / administrative status of a network interface."""

    Up = "Up"
    Down = "Down"
    Testing = "Testing"
    Unknown = "Unknown"

    @classmethod
    def from_u32(cls, value: int) -> InterfaceStatus | str:
        if value == 1:
            return cls.Up
        elif value == 2:
            return cls.Down
        elif value == 3:
            return cls.Testing
        else:
            return f"Unknown({value})"

    @classmethod
    def from_int(cls, value: int) -> InterfaceStatus | str:
        return cls.from_u32(value)


class Interface(BaseModel):
    """Network device physical/logical interface information."""

    model_config = ConfigDict(arbitrary_types_allowed=True, use_enum_values=True)

    index: int
    description: str | None = None
    mac_address: str | None = None
    admin_status: InterfaceStatus | str | None = None
    oper_status: InterfaceStatus | str | None = None
